src/Components/Server/src/Circuits/RemoteJSDataStream.cs

```diff
@@ -0,0 +1,188 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Pipelines;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.JSInterop;
+
+namespace Microsoft.AspNetCore.Components.Server.Circuits
+{
+    internal class RemoteJSDataStream : Stream
+    {
+        // Concerns with static `Instances`? Malicious actor could hijack another user's
+        // stream by (improbably) guessing the GUID.
+        // Maybe we put this in the JSRuntime for curcuit isolation? Would increate overhead as
+        // we'd have to go through CircuitHost, DotNetDispatcher and so on.
+        private readonly static Dictionary<Guid, RemoteJSDataStream> Instances = new();
```

Comment:
I was planning to move this to JS Runtime but that's easier said than done (as `SupplyData` is static and doesn't have access to the `JSRuntime`). Consequently, we'd have to go through the `CurcuitHost` and so on. This would have additional overhead (especially if we need to wait on the [Renderer](https://github.com/dotnet/aspnetcore/blob/1048a02802897ca86de536664db3a30f5b4a9ed7/src/Components/Server/src/Circuits/CircuitHost.cs#L342). I still think it's worthwhile but wanted to double check.

Reply:
Did it via b35027b944. To confirm, we don't have to use the [Renderer](https://github.com/dotnet/aspnetcore/blob/1048a02802897ca86de536664db3a30f5b4a9ed7/src/Components/Server/src/Circuits/CircuitHost.cs#L342), correct?

Reply:
Looks good to me